packages/react-dom/src/FadeIn.tsx

```diff
@@ -0,0 +1,34 @@
+import { type CSSProperties, type ComponentPropsWithoutRef, type ElementType } from 'react'
+import { useInView } from './useInView'
+
+type FadInBaseProps = {
+  delay?: number
+  duration?: number
+  timingFunction?: CSSProperties['animationTimingFunction']
+}
+type FadeInProps<TAs extends ElementType> = {
+  as?: TAs
+} & Omit<ComponentPropsWithoutRef<TAs>, 'as' | keyof FadInBaseProps> &
+  FadInBaseProps
```

Comment:
What do you think about Using `Omit<CSSProperties, 'opacity' | 'transition'>` for type safety?

Reply:
Sorry! I couldn't understand your comment. could you more explain implementation

Reply:
I noticed that since we spread rest props before applying the component's style, users can't override styles currently. We might want to consider merging styles after spreading rest props.

Additionally, if users add `opacity` or `transition` to the style prop, it might break the component's fade-in behavior (controlled by delay and duration props). Using `Omit` would prevent such cases at the type level.

Reply:
Could you commit and push? I want to know your thought clearly

Reply:
I thought about changing it like this

```ts
type FadeInProps<TAs extends ElementType> = Omit<ComponentPropsWithoutRef<TAs>, 'style'> & { // 👈
  /**
   * The element type to render.
   */
  as?: TAs;
  /**
   * The delay in milliseconds before the animation starts.
   */
  delay?: number;
  /**
   * The duration in milliseconds of the animation.
   */
  duration?: number;
  /**
   * The timing function of the animation.
   */
  timingFunction?: CSSProperties['animationTimingFunction'];
  inViewOptions?: InViewOptions;
  style?: Omit<CSSProperties, 'opacity' | 'transition'>; // 👈
};

export function FadeIn<TAs extends ElementType = 'div'>({
  as,
  delay = 0,
  duration = 200,
  timingFunction = 'linear',
  inViewOptions,
  style,
  ...rest
}: FadeInProps<TAs>) {
  const Component = as ?? 'div';
  const { inView, ref } = useInView(inViewOptions);

  return (
    <Component
      {...rest}
      ref={ref}
      style={{
        opacity: inView ? 1 : 0,
        willChange: 'opacity',
        transition: `opacity ${duration}ms ${timingFunction} ${delay}ms`,
        ...style, // 👈
      }}
    />
  );
}
```

Reply:
I think omitting the style type might be unnecessary, but we need to consider allowing users to add their custom styles.

Reply:
I agree! I reflect your comment in [d288710](https://github.com/toss/suspensive/pull/1362/commits/d288710ad2ea1a5433f273b0b22831e42480d480)